Sources/Features/DappsAndPersonas/DappDetails/DappDetails.swift

```diff
@@ -344,16 +336,23 @@ public struct DappDetails: Sendable, FeatureReducer {
 
 	/// Loads any other dApps associated with the dApp
 	private func loadDapps(
-		metadata: GatewayAPI.EntityMetadataCollection,
+		metadata: OnLedgerEntity.Metadata,
 		validated dappDefinitionAddress: DappDefinitionAddress
 	) async -> Loadable<[OnLedgerEntity.AssociatedDapp]> {
-		let dAppDefinitions = try? metadata.dappDefinitions?.compactMap(DappDefinitionAddress.init)
-		guard let dAppDefinitions else { return .idle }
-
-		let associatedDapps = await dAppDefinitions.parallelMap { dApp in
-			try? await extractDappInfo(for: dApp, validating: dappDefinitionAddress)
+		guard let dAppDefinitions = metadata.dappDefinitions else { return .idle }
+
+		let loadedDApps = await (try? onLedgerEntitiesClient.getAssociatedDapps(dAppDefinitions)) ?? []
+		let associatedDapps = loadedDApps.filter { dApp in
+			do {
+				try dApp.metadata.validate(dAppDefinitionAddress: dApp.address)
+				guard let name = dApp.metadata.name else {
+					throw OnLedgerEntity.Metadata.MetadataError.missingName
```

Comment:
this `throw OnLedgerEntity.Metadata.MetadataError.missingName` here will just be caught by the `catch` below, so we are basically suppressing that throw, instead you can `loggerGlobal.notice("missing name"); return false` ?